cherrypy/lib/encoding.py

```diff
@@ -412,7 +415,13 @@ def gzip(compress_level=5, mime_types=['text/html', 'text/plain'],
                 cherrypy.log('Gzipping', context='TOOLS.GZIP')
             # Return a generator that compresses the page
             response.headers['Content-Encoding'] = 'gzip'
-            response.body = compress(response.body, compress_level)
+            # Check if there exists a pre-compressed file, and use it.
```

Comment:
I've got concerns about such behavior being unobvious. I think such things must be explicitly configured.